homeassistant/components/recollect_waste/diagnostics.py

```diff
@@ -0,0 +1,27 @@
+"""Diagnostics support for ReCollect Waste."""
+from __future__ import annotations
+
+import dataclasses
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "entry": {
+            "title": entry.title,
+            "data": dict(entry.data),
+            "options": dict(entry.options),
+        },
```

Comment:
You could simple add the whole config entry and keep in mind to redact sensitive data!
Example:
```python
"entry": async_redact_data(config_entry.as_dict(), REDACT_CONFIG)
```
_use `async_redact_data` from `homeassistant.components.diagnostics`_

Reply:
Good call on using the entire entry! There isn't any sensitive data here to worry about.

Reply:
not sure how wide the place_id is, but this could be seen as sensitiv/private data? 🤔

Reply:
It's very wide (represents a whole city subdivision) and can't be used to identify a specific location like latitude and longitude can. That said, if that still sounds too specific, I'm open!